### PR TITLE
Lvnod 80 guest controller

### PR DESCRIPTION
**Pull request by AlbaMontis**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-80)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-86)

## Code reviewers

- [x] @yurii-kokhalevych
- [x] @shyyka
- [x] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: server/controllers/auth.controller.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const activUser = require('../../config/activUserConfig.json').activUser;
 module.exports = {
   login(req, res) {
     User.findOne({where: {email: req.body.email}}).then(user => {
-      if (!user || !passwordHash.verify(req.body.password, user.password)) {
+      if (!user || !passwordHash.verify(req.body.password, user.password) || !user.is_activate) {
         res.status(401).send('Email or password is not valid');
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: why if !user.is_activate response Email or password is not valid?

File: server/controllers/guest.controller.js [hunk 2]
```diff
@@ -4,63 +4,66 @@ const Guest = require('../../config/db').Guest;
...
       where: {event_id: req.params.id},
-      include: [User]
+      include: [{
+        model: User,
+        attributes: ['id', 'email']
+      }]
     })
     .then(guests => {
-      res.send({guests: guests});
+      res.send({guests});
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: add status 200

File: server/helper/mailer.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ module.exports = {
       }
 
       let mailOptions = {
-        from: '"Da-Mi"' + secret.gmail.user,
+        from: '"Da-Mi"' + '<' + secret.gmail.user + '>',
```
Comment (4asof4ik) [inline server/helper/mailer.js]: look at the line 25  use `${}`

File: server/controllers/user.controller.js [hunk 4]
```diff
@@ -95,25 +113,50 @@ module.exports = {
...
-      if (!user) {
-        return res.status(404).send(messages.userNotFound);
-      }
-      return user
-      .destroy()
-      .then(user => res.status(204).send(user))
-      .catch(error => res.status(404).send(messages.userNotFound));
-    })
-    .catch(error => res.status(404).send(messages.userNotFound));
+    const eventId = req.headers.referer.split('/')[4];
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: find in params

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/user.controller.js (hunk 4 above)
Comment (AlbaMontis) [reply server/controllers/user.controller.js]: ok.

File: server/controllers/guest.controller.js (hunk 2 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: ok.

File: server/helper/mailer.js (hunk 3 above)
Comment (AlbaMontis) [reply server/helper/mailer.js]: ok.

File: server/controllers/auth.controller.js (hunk 1 above)
Comment (AlbaMontis) [reply server/controllers/auth.controller.js]: I'll fix it.

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: server/controllers/auth.controller.js [hunk 5]
```diff
@@ -10,15 +9,15 @@ const activUser = require('../../config/activUserConfig.json').activUser;
...
-      if (!user || !passwordHash.verify(req.body.password, user.password)) {
-        res.status(401).send('Email or password is not valid');
-      } else {
-        const token = jwt.sign({
-          id: user.id
-        }, secret.key, {expiresIn: '2h'});
-        res.json({'token': token, 'user_id': user.id});
-      }
+      (!user || !passwordHash.verify(req.body.password, user.password)) &&
+        res.status(401).send(messages.loginDataNotValid) ||  !user.is_activate &&
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: after || 2 spaces

File: server/controllers/auth.controller.js [hunk 6]
```diff
@@ -10,15 +9,15 @@ const activUser = require('../../config/activUserConfig.json').activUser;
 
 module.exports = {
   login(req, res) {
+    let token;
     User.findOne({where: {email: req.body.email}}).then(user => {
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: separat

File: server/controllers/auth.controller.js [hunk 7]
```diff
@@ -10,15 +9,15 @@ const activUser = require('../../config/activUserConfig.json').activUser;
...
-        const token = jwt.sign({
-          id: user.id
-        }, secret.key, {expiresIn: '2h'});
-        res.json({'token': token, 'user_id': user.id});
-      }
+      (!user || !passwordHash.verify(req.body.password, user.password)) &&
+        res.status(401).send(messages.loginDataNotValid) ||  !user.is_activate &&
+          res.status(401).send(messages.userIsNotActivated) ||
+            (token = jwt.sign({
+              id: user.id
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: and 20 line

File: server/controllers/user.controller.js [hunk 8]
```diff
@@ -95,21 +109,46 @@ module.exports = {
...
+            user_id: req.params.id
+          }
+        })
+        .then(guests => {
+          guests.forEach(guest => guest && guest.destroy());
+        });
+      })
+      .then(() => res.status(204).send(messages.userDeleted))
+      .catch(error => {
+        res.status(404).send(messages.userNotFound);
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: in one line without {} check all

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: server/routes/index.js [hunk 9]
```diff
@@ -28,6 +28,8 @@ module.exports = app => {
 
   app.post('/api/event/:id/guest/invite', guestController.invite);
   app.get('/api/event/:id/guest/get', guestController.list);
```
Comment (4asof4ik) [inline server/routes/index.js]: why not :id/guests

File: server/helper/mailer.js [hunk 10]
```diff
@@ -42,15 +42,15 @@ module.exports = {
...
       }
 
-      let mailOptions = {
-        from: '"Da-Mi"' + secret.gmail.user,
+      const mailOptions = {
+        from: `"Da-Mi"<${secret.gmail.user}>`,
         to: _data.email,
         subject: _data.subject,
         html: sendMail.html,
         text: sendMail.text,
```
Comment (4asof4ik) [inline server/helper/mailer.js:52]: It should be html or text, Check what is sendMail.text, please

File: server/controllers/auth.controller.js [hunk 11]
```diff
@@ -10,16 +9,21 @@ const activUser = require('../../config/activUserConfig.json').activUser;
...
+    let token;
+    let id;
+    User.findOne({
+      where: {
+        email: req.body.email
       }
     })
+    .then(user => {
+      (!user || !passwordHash.verify(req.body.password, user.password)) &&
+        res.status(401).send(messages.loginDataNotValid) || !user.is_activate &&
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: why you add spaces in each line

File: server/controllers/guest.controller.js [hunk 12]
```diff
@@ -1,66 +1,67 @@
...
 
-const Guest = require('../../config/db').Guest;
-const User = require('../../config/db').User;
-const Event = require('../../config/db').Event;
-const Profile = require('../../config/db').Profile;
-const mailer = require('./../helper/mailer');
-const URL = require('./../helper/constants');
+const { Guest, User, Event, Profile } = require('../../config/db');
+const { mailer, constants, messages } = require('./../helper');
+const URL = constants.HOST + constants.PORT;
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: create this in const file

File: server/controllers/guest.controller.js [hunk 13]
```diff
@@ -1,66 +1,67 @@
...
       where: {event_id: req.params.id},
-      include: [User]
+      include: [{
+        model: User,
+        attributes: ['id', 'email']
+      }]
     })
     .then(guests => {
-      res.send({guests: guests});
+      res.status(200).send({guests});
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js]: drop {} and into line 17

File: server/controllers/user.controller.js [hunk 14]
```diff
@@ -91,25 +103,41 @@ module.exports = {
...
+      User.findById(req.params.id)
+      .then(user => {
+        user && user.destroy();
+        Guest.findAll({
+          where: {
+            user_id: req.params.id
+          }
+        })
+        .then(guests => {
+          guests.forEach(guest => guest && guest.destroy());
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: drop {}

File: server/controllers/user.controller.js [hunk 15]
```diff
@@ -1,44 +1,56 @@
...
-          Guest.findOne({
-            where: {
-              user_id: user.id
-            }
-          })
-          .then(guest => {!guest &&
+          findGuest(user.id, eventId)
+          .then(guest => {
+            !guest &&
             Guest.create({
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: find or create maybe without findGuest(user.id, eventId) ?

File: server/controllers/user.controller.js [hunk 16]
```diff
@@ -1,44 +1,56 @@
...
               event_id: eventId,
               user_id: user.id
             });
           });
-        });
-      }) :
+          return {id: user.id, email: user.email};
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: this is arrow function return is already included

File: server/controllers/user.controller.js [hunk 17]
```diff
@@ -91,25 +103,41 @@ module.exports = {
...
+        })
+        .then(() => res.status(204).send(messages.guestDeleted))
+        .catch(error => res.status(404).send(messages.guestNotFound)) ||
+          res.status(403).send(messages.accessDenied);
+      }) ||
+
+      User.findById(req.params.id)
+      .then(user => {
+        user && user.destroy();
+        Guest.findAll({
```
Comment (4asof4ik) [inline server/controllers/user.controller.js]: use cascade in the relationship  when you destroy user all guests with userId will be destroyed also

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/routes/index.js (hunk 9 above)
Comment (AlbaMontis) [reply server/routes/index.js]: ok.

File: server/controllers/user.controller.js (hunk 17 above)
Comment (AlbaMontis) [reply server/controllers/user.controller.js]: ok

File: server/controllers/guest.controller.js (hunk 13 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: ok.

File: server/controllers/user.controller.js (hunk 15 above)
Comment (AlbaMontis) [reply server/controllers/user.controller.js]: ok

File: server/controllers/user.controller.js (hunk 14 above)
Comment (AlbaMontis) [reply server/controllers/user.controller.js]: I'll delete this row due to previous comment.

File: server/controllers/auth.controller.js (hunk 11 above)
Comment (AlbaMontis) [reply server/controllers/auth.controller.js]: Fixed.

File: server/controllers/guest.controller.js (hunk 12 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js]: ok.

**Review by shyyka — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

File: server/controllers/auth.controller.js [hunk 18]
```diff
@@ -10,16 +9,21 @@ const activUser = require('../../config/activUserConfig.json').activUser;
...
-        res.json({'token': token, 'user_id': user.id});
+    let token;
+    let id;
+    User.findOne({
+      where: {
+        email: req.body.email
       }
     })
+    .then(user => {
+      (!user || !passwordHash.verify(req.body.password, user.password)) &&
```
Comment (4asof4ik) [inline server/controllers/auth.controller.js]: can you create function  which will be check (!user || !passwordHash.verify(req.body.password, user.password))  and return true or false

File: server/controllers/guest.controller.js [hunk 19]
```diff
@@ -1,66 +1,64 @@
...
-          email: guest.User.email,
-          eventName: guest.Event.name,
-          date: guest.Event.date_event,
-          eventDescription: guest.Event.description,
-          img: 'party.jpg'
-        }, template);
-      });
-      return guests;
+          mailer({
+            host: constants.URL,
```
Comment (4asof4ik) [inline server/controllers/guest.controller.js:45]: in production it will be localhost:8080 it's okey?

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: server/controllers/guest.controller.js (hunk 19 above)
Comment (AlbaMontis) [reply server/controllers/guest.controller.js:45]: It will not.

File: server/controllers/auth.controller.js (hunk 18 above)
Comment (AlbaMontis) [reply server/controllers/auth.controller.js]: Done.

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yuliannakorol — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by shyyka — APPROVED** [review]

**Review by yurii-kokhalevych — APPROVED** [review]

**Review by yuliannakorol — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**